docs/rfc/0001-exporter-interface.md

```diff
@@ -0,0 +1,194 @@
+following components:
+* An implementation of the exporter interface.
+```
+// Exporter defines the methods invoked during the plugin lifecycle of a data exporter.
+type Exporter interface {
+  // Connect will be called during initialization, before block data starts going through the pipeline.
+  // Typically used for things like initializating network connections.
+  // The ExporterConfig passed to Connect will contain the Unmarhsalled config file specific to this plugin.
+  // Should return an error if it fails--this will result in the Indexer process terminating.
+  Connect(ExporterConfig cfg) error
+  
+  // Shutdown will be called during termination of the Indexer process.
+  // There is no guarantee that plugin lifecycle hooks will be invoked in any specific order in relation to one another.
+  // Returns an error if it fails which will be surfaced in the logs, but the process is already terminating.
+  Shutdown() error
+  
+  // Recv is called for each block to be processed by the exporter.
+  // Should return an error on failure--retries are configurable.
+  Recv(DataFormat blockData) error
+  
+  // DataFormat returns the type of the data that the plugin expects to receive.
+  // Must be one of the impelemented formats in the Indexer repository.
+  DataFormat() DataFormat
+  
+  // Round returns the next round to be processed. Atomically updated when Recv successfully completes.
+  Round() uint64
+}
+```
+
+***Plugin Config***
+* A config file that defines all parameters that can be supplied to the plugin, and which provides the default values
+that will be used for each parameter. A toml file stored inside the Indexer data directory stores all of the config data
+for a given plugin. The Indexer will use the type specified in the Indexer config to look for a plugin config which
+satisfies that class, and then load that as the selected plugin. Supplying multiple plugin configs for the selected type
+of exporter will result in undefined behavior--a random config will be chosen.
+```
+[Plugin.Exporter]
+name = "postgresql-exporter"
+type = "IndexerPostgresqlExporter"
+username = "foo"
+password = "bar"
+host = "127.0.0.1"
+port = "1234"
+dbname = "indexer"
+```
+
+**Indexer Config**
+
+The Indexer's config will need to be changed to incorporate plugins. Future RFCs can decide how to string multiple
+pipelines and plugins together in interesting ways. For now, we will have a single pipeline per indexer--i.e. running the
+indexer will only run a single data pipeline and therefore have a single exporter. In that way the initial changes will
+be easy to make--one new field in the config which provides an internal plugin name. Config files for the selected
+plugin will then be resolved and parsed at runtime. The default configuration for Indexer will select the Postgresql
+plugin in order to maintain backwards compatibility with the existing Indexer pipeline.
```

Comment:
I really like this idea, nice way to phase in plugins. Maybe we can call this setting "exporter override"?